mon-pix/app/controllers/campaigns/skill-review.js

```diff
@@ -1,31 +1,30 @@
 import Controller from '@ember/controller';
+import { computed } from '@ember/object';
 
 export default Controller.extend({
 
-  showButtonToShareResult: true,
   displayLoadingButton: false,
   displayErrorMessage: false,
 
+  showButtonToShareResult: computed('model.campaignParticipation.isShared', function() {
+    return !this.get('model.campaignParticipation.isShared');
+  }),
+
   actions: {
     shareCampaignParticipation() {
       this.set('displayErrorMessage', false);
       this.set('displayLoadingButton', true);
       const campaignParticipation = this.get('model.campaignParticipation');
       campaignParticipation.set('isShared', true);
-      campaignParticipation.save()
-        .then(function() {
-          this.set('showButtonToShareResult', false);
+      return campaignParticipation.save()
+        .then(() => {
           this.set('displayLoadingButton', false);
```

Comment:
malgré la suppression du `bind(this)`, ce `this` a toujours la bonne référence et set le bon truc ? 🤔

Reply:
L'écriture `() => {}` fait un `bind` automatique du `context` courant (contrairement à l'écriture `function() {}`

Reply:
AH ! Je savais pas

Reply:
ou plutôt, ça me dit quelque chose, j'ai dû le savoir, mais oublier ^^